### PR TITLE
feat: (Core) introduce Object number component

### PR DESCRIPTION
**Pull request by IlyaSurmay**
#### Please provide a link to the associated issue.

closes https://github.com/SAP/fundamental-ngx/issues/3284

#### Please provide a brief summary of this pull request.

Introduce core object number component

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]
LGTM

File: libs/core/src/lib/object-number/object-number.component.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        this.onChanges();
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.onChanges();
+    }
+
+    /** @hidden */
+    onChanges(): void {
```
Comment (Betrozov) [inline libs/core/src/lib/object-number/object-number.component.ts:84]: seems it should be private

File: libs/core/src/lib/object-number/object-number.component.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        return [
+            'fd-object-number',
+            this.large ? 'fd-object-number--large' : '',
+            this.status ? `fd-object-number--${this.status}` : '',
+            this.class
+        ];
+    }
+
+    /** @hidden */
+    buildNumberPipeConfig(): void {
```
Comment (Betrozov) [inline libs/core/src/lib/object-number/object-number.component.ts:104]: the same

File: libs/core/src/lib/object-number/object-number.component.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        ];
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    /** @hidden */
+    private onChanges(): void {
```
Comment (Betrozov) [inline libs/core/src/lib/object-number/object-number.component.ts:103]: all private methods should be with the prefix `_`

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Review by mykolavorobiov — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Pull request merged**